server/src/inmantals/server.py

```diff
@@ -64,9 +65,149 @@ def __init__(self, message: str) -> None:
+            with open(os.path.join(self.get_folder_path(), ".vscode", "settings.json")) as settings_file:
+                settings: dict[str, str] = json.load(settings_file)
+                return settings.get(key, default)
+        except FileNotFoundError:
+            return default
+
+    @classmethod
+    def unpack_workspaces(cls, workspace_folders: Sequence[object], ls_handler: "InmantaLSHandler") -> Dict[str, "Folder"]:
+        return {folder["uri"]: Folder(URI(folder["uri"]), folder["name"], ls_handler) for folder in workspace_folders}
+
+    def cleanup(self) -> None:
+        logger.info(f"calling cleanup for {self}")
+        if self.inmanta_project_dir:
+            self.handler.remove_folder(str(self.folder_uri))
+            self.inmanta_project_dir.cleanup()
+
+    def get_folder_path(self) -> str:
+        """
+        Convert the folders's uri into a regular path
+        (https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#uri)
+        """
+        return self.folder_uri.path
+
+    def create_tmp_project(self) -> str:
+        """
+        When working on a module, an inmanta project is created in a temporary directory.
+        """
+        self.inmanta_project_dir = tempfile.TemporaryDirectory()
+        logger.debug(f"Temporary project created at {self.inmanta_project_dir.name}.")
+
+        os.mkdir(os.path.join(self.inmanta_project_dir.name, "libs"))
+
+        install_mode = module.InstallMode.master
+
+        v2_metadata_file: str = os.path.join(self.get_folder_path(), module.ModuleV2.MODULE_FILE)
+        v1_metadata_file: str = os.path.join(self.get_folder_path(), module.ModuleV1.MODULE_FILE)
+
+        module_name: Optional[str] = None
+        libs_folder = os.path.join(self.inmanta_project_dir.name, "libs")
+
+        if os.path.exists(v1_metadata_file):
+            mv1 = module.ModuleV1(project=None, path=self.get_folder_path())
+            module_name = mv1.name
+            os.symlink(self.get_folder_path(), os.path.join(libs_folder, module_name), target_is_directory=True)
+
+        elif os.path.exists(v2_metadata_file):
+            mv2 = module.ModuleV2(project=None, path=self.get_folder_path(), is_editable_install=True)
+            module_name = mv2.name
+            # Install this v2 module in editable mode in the active venv.
+            env.process_env.install_from_source([env.LocalPackagePath(path=self.get_folder_path(), editable=True)])
+
+        if not module_name:
+            error_message: str = (
+                "The Inmanta extension only works on projects and modules. "
+                f"Please make sure the folder opened at {self.get_folder_path()} is a valid "
+                "[project](https://docs.inmanta.com/inmanta-service-orchestrator/latest/model_developers/project_creation.html)"
+                " or "
+                "[module](https://docs.inmanta.com/inmanta-service-orchestrator/latest/model_developers/module_creation.html)"
```

Comment:
This is a though one but I think we have to refer to the OSS docs (https://docs.inmanta.com/community/latest/) because not everyone can necessarily access these.

Reply:
True, nice catch